### PR TITLE
refactor: pull out `.length` from for loop

### PR DESCRIPTION
**Pull request by movahhedi**
On something like the router, the smallest improvement makes a difference.

https://dev.to/siddiqus/which-for-loop-is-the-fastest-in-javascript-4hdf

### CODE REVIEW

File: src/router/linear-router/router.ts [hunk 1]
```diff
@@ -20,7 +20,8 @@ export class LinearRouter<T> implements Router<T> {
 
   match(method: string, path: string): Result<T> {
     const handlers: [T, Params][] = []
-    ROUTES_LOOP: for (let i = 0, len = this.routes.length; i < len; i++) {
+    const routesLength = this.routes.length
+    ROUTES_LOOP: for (let i = 0, len = routesLength; i < len; i++) {
```
Comment (EdamAme-x) [inline src/router/linear-router/router.ts:24]:
```suggestion
    const routesLength = this.routes.length
    ROUTES_LOOP: for (let i = 0; i < routesLength; i++) {
```

Is this not good enough?

**Comment by yusukebe** [comment]
Hi @movahhedi 

These are the same behavior. It does not calculate the length every time in the current code.

**Comment by movahhedi** [comment]
Oh my God, my eyes don't seem to see any more. 🤦‍♂️
I was looking at it in the GitHub web code viewer.
Sorry for the waste of time.

**Comment by yusukebe** [comment]
@movahhedi No problem!!

**Pull request closed without merging**